### PR TITLE
Fix docs service and add RTD support

### LINKED ISSUE #2733: RTD Support
## Description

ReadTheDocs support

## Rationale

Hosting an open-source website with documentation 

Also somewhat of a bug report because you need a .readthedocs.yml (I think) file in order to pip install the local.txt packages.

## Use case(s) / visualization(s)

Developers for an open-source (or closed-source) website can easily read up on how the website functions.

### PR DESCRIPTION
**Pull request by Andrew-Chen-Wang**
Fixes #2733 
Closes #2878 
Closes #2674

Supersedes #2754

## Description

Fix docs service. Added `DATABASE_URL` environment variable inside `conf.py` to force `django.setup()` in both local and production environments to use a SQLite database. Also stays consistent with setting the other entrypoint env var, `CELERY_BROKER_URL`.

Checklist:

- [X] I've made sure that `tests/test_cookiecutter_generation.py` is updated accordingly (especially if adding or updating a template option)
- [X] I've updated the documentation or confirm that my change doesn't require any updates

## Rationale

Fixes doc service and adds doc option for production.

### CODE REVIEW

**Comment by Andrew-Chen-Wang** [comment]
Sorry for the large number of commits. I must've messed up my git command or something.

File: {{cookiecutter.project_slug}}/docs/_source/conf.py [hunk 1]
```diff
@@ -14,12 +14,20 @@
...
+    os.environ["DJANGO_READ_DOT_ENV_FILE"] = "True"
+    {%- if cookiecutter.use_celery == 'y' %}
+    os.environ["CELERY_BROKER_URL"] = os.getenv("REDIS_URL", "redis://redis:6379")
+    {%- endif %}
+    os.environ["USE_DOCKER"] = "no"
+else:
+{%- if cookiecutter.use_docker == 'y' %}
+    sys.path.insert(0, os.path.abspath("/app"))
+{%- else %}
+    sys.path.insert(0, os.path.abspath("../.."))
```
[the remark this answers is not in the document]
Comment (Andrew-Chen-Wang) [reply {{cookiecutter.project_slug}}/docs/_source/conf.py]: Ah sorry I didn't test this out locally. Completely my fault for not catching that.

Having everything inside the `/_source` made everything organizationally better; I'm just wondering if there's any other option... although, I think I'd have to agree that moving everything back to `docs/` may just be the only solution.

[the remark this answers is not in the document]
Comment (browniebroke) [reply {{cookiecutter.project_slug}}/docs/_source/conf.py]: We do this in the template documentation and it's not too bad IMO. It's probably possible to make it work inside or `_source` but it feels easier being outside and I'm not sure I'm willing to invest more time in this at this point.

**Review by browniebroke — APPROVED** [review]
I believe this is now ready to merge, thanks for the effort @Andrew-Chen-Wang

**Comment by Andrew-Chen-Wang** [comment]
Thanks for the review!

**Pull request merged**